standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -662,7 +655,7 @@ public void rollbackTransaction() {
         currentTransaction.rollback();
       }
     } finally {
-      openTrasactionCalls = 0;
+      openTrasactionCalls--;
```

Comment:
afyter rollback openTrasactionCalls should be 0

Reply:
Our unit test could not pass if is 0 here, the key point seems that if we should support such usage:
```bash
openTransaction()
  // new function1
  {
    openTransaction()
    commitTransaction() # place1
  }

  // new function2
  {
    openTransaction()
    commitTransaction()
  }
commitTransaction() # place2
```